xcat-inventory/xcclient/inventory/backend.py

```diff
@@ -0,0 +1,695 @@
+#!/usr/bin/python
+from __future__ import print_function
+import 	configparser
+import os
+import sh
+import shutil
+import re
+import sys
+from exceptions import * 
+import manager 
+import utils
+import pickle
+import hashlib
+
+class Invbackend:
+
+    globalcfgpath="/etc/xcat/inventory.cfg"
+    bkendcfg={}
+
+    def loadcfg(self,cfgpath=None):
+        if cfgpath is None:
+            myhome=utils.gethome()
+            cfgpath=myhome+'/.xcatinv/inventory.cfg' 
+        if not os.path.isfile(cfgpath):
+            cfgpath=self.globalcfgpath 
+
+        if not os.path.exists(cfgpath):
+            raise InvalidFileException("The configuration file %s does not exist!\n"%(cfgpath))    
+
+        config = configparser.ConfigParser()
+        config.read(cfgpath)
+        if not config:
```

Comment:
Looks like you have wrong order of L31 and L32-L33

Reply:
config will not be None,  you should check `'backend' in config.sections()` only.